### ISSUE TITLE
Image not loaded for first ROM on the list

### ISSUE THREAD

**Issue by IyonUK** [RetroPie/EmulationStation#185]
If I enter a rom list and the first rom has an image then it doesn't show immediately. However, if I move to the next rom on the list and back again the image appears.

**Comment by hex007** [comment]
Please share :
- Platform
- ES version
- ES binary source (if not stock)
- Gamelist which has problems

**Comment by pjft** [comment]
This is known to happen in sub folders only. Please confirm that's the
case.

**Comment by IyonUK** [comment]
Hi

Sorry, AFK at the moment. It’s the version of Emulationstation bundled with RetroPie v4.2 running on Raspberry Pi 3. But, as pjft says, I am using subfolders and that’s where it’s happening (they are all in subfolders).

Sorry, couldn’t see an issue for this but maybe it’s on the original fork.

Stephen

**Comment by pjft** [comment]
Thanks Stephen.

It's a long-standing bug, thanks for reporting it though. It's good to keep
a record.


- Paulo Tavares

**Comment by pjft** [comment]
I've fixed this on the current build for adding collections in sub-folders, so this should be fixed when it gets submitted and merged.

**Comment by pjft** [comment]
Hi. 

Please update to the latest ES and see if it's still happening. You may have to install from source, as I'm not sure the binary is already available. 

Thanks.

**Comment by zigurana** [comment]
Hey @pjft, which change did fix this issue? I see the same thing in my
refactoring branch, but then also in cases without subfolders, so I was
curious how you fixed it.

**Comment by pjft** [comment]
Not a bother. I _think_ (it was a few weeks back) that it was with what I
just commented you in.

If you don't receive the notification, I added

FileData* cursor = getCursor();
setCursor(cursor);

to the ISimpleGameListView::input(...) method, when opening the folder,
after

mCursorStack.push(cursor);
populateList(cursor->getChildrenListToDisplay());

As setCursor is what causes the view elements to be updated with the cursor
metadata. We had it when we pressed "b" to go up a level (when in a folder)
but not the other way around.

Feel free to add it on your end and see if it helps, but it should be it.


- Paulo Tavares
